pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -944,6 +945,15 @@ func (l *LibvirtDomainManager) migrate(vmi *v1.VirtualMachineInstance, options *
 		return
 	}
 
+	// finalize the migration here. MigrateToURI3 is synchronic. If we got here
+	// it means that the migration has successfully completed. Due to its
+	// nature, it is very likely that the migration monitor will miss
+	// DOMAIN_JOB_COMPLETED which quickly switches to DOMAIN_JOB_NONE.
+	// Migration monitor will not have enough time to find the correct result
+	// and it will be in a race with the domain shutdown proceess, which starts
+	// immediately after a successful migration.
+
+	l.setMigrationResult(false, "", "")
```

Comment:
This is great but as part of this PR can we simplify the migrationMonitor code? Now the branching is a bit worse and there is a race between this line and the migration monitor, as far as I can tell everything is correct but logs can get a bit confusing.

For example:
1. migration completes
2. migrate() sets the migration result
3. the migration monitor sees DOMAIN_JOB_NONE
4. determineNonRunningMigrationStatus() gets called and logs `logger.Info("Migration job didn't start yet")`

This would be confusing as you would see something like
```
Live migration succeeded.
Migration job didn't start yet
```

Maybe we can have the migration monitor just monitor abortion and cancellation since `MigrateToURI3` is sync and will return signaling either failure or completion, what do you think?